File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, m = map(int,input().split())
numbers = list(map(int,input().split()))
result = 0

for i in range(n) :
    for j in range(i + 1, n) :
        for k in range(j + 1 , n) :
            n_sum = numbers[i] + numbers[j] + numbers[k]
            if n_sum <= m :
                # 최대값 비교
                result = max(result, n_sum)

print(result)

File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
using namespace std;

int main(){
	int N, M, result = 0;
	cin >> N >> M;
	
	int nums[N];
	for(int i=0; i<N; i++){
		cin >> nums[i];
	}
	
	for(int i=0; i<N; i++){
		for (int j=i+1; j<N; j++){
			for (int k=j+1; k<N; k++){
				int n_sum = nums[i] + nums[j] + nums[k];
				
				if (n_sum <= M && n_sum > result){
					result = n_sum;
				}
			}
		}
	}
	
	cout << result;
	
	return 0;
}